File: main_module/src/main/java/com/phone/main_module/login/view/ILoginView.java
package com.phone.main_module.login.view;

import com.phone.common_library.base.IBaseView;
import com.phone.main_module.login.bean.GetVerificationCode;
import com.phone.main_module.login.bean.LoginResponse;

public interface ILoginView extends IBaseView {
	
	void getAuthCodeSuccess(GetVerificationCode.DataDTO success);
	
	void getAuthCodeError(String error);
	
	void loginWithAuthCodeSuccess(LoginResponse.DataDTO success);
	
	void loginWithAuthCodeError(String error);

	void loginSuccess(String success);

	void loginError(String error);

}

File: main_module/src/main/java/com/phone/main_module/login/view/ILoginView.kt
package com.phone.main_module.login.view

import com.phone.common_library.base.IBaseView
import com.phone.main_module.login.bean.GetVerificationCode
import com.phone.main_module.login.bean.LoginResponse

interface ILoginView:IBaseView {

    fun getAuthCodeSuccess(success: GetVerificationCode.DataDTO?)

    fun getAuthCodeError(error: String?)

    fun loginWithAuthCodeSuccess(success: LoginResponse.DataDTO?)

    fun loginWithAuthCodeError(error: String?)

    fun loginSuccess(success: String?)

    fun loginError(error: String?)

}